### ISSUE TITLE
[styles] Represent no-exit tags in dead-end streets.

### ISSUE THREAD

**Issue by patepelo** [organicmaps/organicmaps#7892]
Tag: [noexit=yes](https://wiki.openstreetmap.org/wiki/Tag:noexit%3Dyes)
Uses: 1.7M

Ideas on how to represent? 

This is the current Id Editor approach:

![imagen](https://github.com/organicmaps/organicmaps/assets/23003127/9bb860ae-ae48-464d-a29a-847c5b0769d7)

### Discussed in https://github.com/orgs/organicmaps/discussions/7890

<div type='discussions-op-text'>

<sup>Originally posted by **justwalked** April 11, 2024</sup>
I would like to suggest rendering [noexit=yes](https://wiki.openstreetmap.org/wiki/Tag:noexit%3Dyes) nodes to let the user better understand which ways can and cannot be used to navigate.

Allow me to share my own story that led me writing this suggestion. I have been using OrganicMaps to navigate around Taiwan by walk. While not being perfect, OSM data is quite good in Taiwan and OrganicMaps does a good job at calculating itineraries, however I came across some unconnected ways or outdated information that made OrganicMaps route around entire areas I could have otherwise walked through. I soon learned that the data is not 100% perfect, especially in remote areas, so I have compensate for map limitations by guessing and asking locals.

So at some point I came across an area that can seemingly be crossed by two possible ways, either a small or a big road, but the small road is not connected so OrganicMaps always routes through the main road no matter what. As a pedestrian I wish I could take the small road to avoid dangerous traffic. So should I assume the way is incorrectly unconnected and go the small road? If it happens to be a dead end, I have to go back all the way, which means walking an extra 3.4 km for nothing, which is so frustrating for a pedestrian.

![map](https://github.com/organicmaps/organicmaps/assets/166682256/524ceea7-f6d6-4536-a2a1-d63b34ab7d1e)

[Link of the map above](https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=22.72615%2C120.25867%3B22.72857%2C120.27195#map=17/22.72868/120.26586&layers=C)

On that day, I did not want to take risks so I walked the main road, and after crossing I saw that the small road seemed to be blocked by a big fence. I thought thank god I did not take the small road! Later, I wanted to edit the map to make it clearer that the small road is a dead end by adding a fence or something. To my surprise, I discovered there is a [noexit](https://www.openstreetmap.org/node/8512750885) attribute at the road end (also learned about noexit attribute), which is exactly the piece of information I *really* wished I had when I was on the road.</div>

**Comment by biodranik** [comment]
> This is the current Id Editor approach:

Not clear :( A recognizable dead-end icon can be helpful. Or just an X symbol (it will print something like "A dead end without exit" on selection.

**Comment by map-per** [comment]
Interesting idea.
What we should consider, however, is that the `noexit=yes` is not usually rendered by maps, but mostly an OSM internal info for other mappers and for quality assurance. As this tag is not intended to be used by data consumers `noexit=yes` is not tagged on every dead end. It might, therefore, be confusing for OM users that some dead ends have `noexit=yes` and some don't.

**Comment by patepelo** [comment]
> As this tag is not intended to be used by data consumers `noexit=yes` is not tagged on every dead end. It might, therefore, be confusing for OM users that some dead ends have `noexit=yes` and some don't.

You're right. It's not intended for end data consumer so I wouldn't make highlight it very much as it may even feel very unfamiliar for the general user.

**Comment by matkoniecz** [comment]
> So at some point I came across an area that can seemingly be crossed by two possible ways, either a small or a big road, but the small road is not connected so OrganicMaps always routes through the main road no matter what. As a pedestrian I wish I could take the small road to avoid dangerous traffic. So should I assume the way is incorrectly unconnected and go the small road? If it happens to be a dead end, I have to go back all the way, which means walking an extra 3.4 km for nothing, which is so frustrating for a pedestrian.

That falls into some deficiencies of automatically generated maps - in traditional mapping you would have generalization.
And road layout on map would not be just 1:N reflection of reality. For example almost-joining roads would be moved so they are clearly shown as not actually joining.

Doing it automatically is horrifically hard.

BTW, in https://en.wikipedia.org/wiki/Cartographic_generalization#Displace there is image (anchored in https://en.wikipedia.org/wiki/Cartographic_generalization#Exaggerate ).

It shows exactly this problem for OSM-based map, see

> In this [OpenStreetMap](https://en.wikipedia.org/wiki/OpenStreetMap) map of [Loveland Pass](https://en.wikipedia.org/wiki/Loveland_Pass), [Colorado](https://en.wikipedia.org/wiki/Colorado), symbol exaggeration of the thickness of the roads has made them run together. Geometric exaggeration of the hairpin turns and displacement of the roads alongside the interstate are needed to clarify the road network.

**Comment by matkoniecz** [comment]
> So should I assume the way is incorrectly unconnected and go the small road? If it happens to be a dead end, I have to go back all the way, which means walking an extra 3.4 km for nothing, which is so frustrating for a pedestrian.

For practical hints: I would use routing to confirm passability. There always can be gate, tiny section or other blocker.

**Comment by RedAuburn** [comment]
closing this for the reasons discussed above
